spec/interactors/schedule/create_interactor_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+RSpec.describe Schedule::CreateInteractor do
+  describe ".call" do
+    before { stub_any_publishing_api_put_intent }
+    let(:edition) { create(:edition, :schedulable) }
+    let(:user) { build(:user) }
+
+    def build_params(document: edition.document, review_status: "reviewed")
+      ActionController::Parameters.new(document: document.to_param,
+                                       review_status: review_status)
+    end
+
+    it "delegates to ScheduleService to schedule the edition for publishing" do
+      publish_time = 3.days.from_now.beginning_of_day
+      edition = create(:edition, :schedulable, proposed_publish_time: publish_time)
+
+      expect(ScheduleService)
+        .to receive(:call) do |schedule_edition, schedule_user, new_scheduling|
+          expect(schedule_edition).to eq(edition)
+          expect(schedule_user).to eq(user)
+          expect(new_scheduling.reviewed).to be(false)
+          expect(new_scheduling.publish_time).to eq(publish_time)
+        end
+
+      params = build_params(document: edition.document,
+                            review_status: "not_reviewed")
+      Schedule::CreateInteractor.call(params: params, user: user)
+    end
+
+    it "creates a timeline entry" do
+      expect { Schedule::CreateInteractor.call(params: build_params, user: user) }
+        .to change { TimelineEntry.where(entry_type: :scheduled).count }
+        .by(1)
+    end
+
+    it "returns a success with edition" do
+      result = Schedule::CreateInteractor.call(params: build_params, user: user)
+
+      expect(result).to be_success
+      expect(result.edition).to eq(edition)
```

Comment:
Should this test also check `publish_time`, as the UpdateInteractor test does?

Reply:
Good catch, it's more that we shouldn't need to test publish_time since it's only needed internally to the class. I've amended this to be that we just test that both succeed by default.